src/syft/core/common/message.py

```diff
@@ -16,23 +19,29 @@
 from syft.decorators.syft_decorator_impl import syft_decorator
 from ..common.serde.deserialize import _deserialize
 
+# this generic type for SignedMessage
+SignedMessageT = TypeVar("SignedMessageT")
 
-class AbstractMessage(ObjectWithID):
+
+class AbstractMessage(ObjectWithID, Generic[SignedMessageT]):
     """"""
 
     # this should be overloaded by a subclass
-    signed_type = None
+    signed_type: Type[SignedMessageT]
 
 
 class SyftMessage(AbstractMessage):
     def __init__(self, address: Address, msg_id: Optional[UID] = None) -> None:
         self.address = address
         super().__init__(id=msg_id)
 
-    def sign(self, signing_key: SigningKey) -> "SignedMessage":
+    def sign(self, signing_key: SigningKey) -> SignedMessageT:
 
         signed_message = signing_key.sign(self.serialize(to_binary=True))
 
+        # signed_type will be the final subclass callee's closest parent signed_type
+        # for example ReprMessage -> ImmediateSyftMessageWithoutReply.signed_type
+        # == SignedImmediateSyftMessageWithoutReply
```

Comment:
These updates look good - i imagine this was to suppress complaints from the pre-commit hook